### PR TITLE
Sleeper rework(?)

### PR DESCRIPTION
**Pull request by MarkSuckerberg**
## About The Pull Request
image
*Default sleeper with roundstart chems list, subject to change*

Makes sleepers draw from an attached chembag full of beakers for their reagents, finally eliminating the infinitly spawning medicines and bad band-aids such as power cells.

## Why It's Good For The Game
Makes sleepers no more powerful than any other way of applying medicine, more or less becoming a stationary hypospray with a large amount of reagents.

## Changelog
:cl:
add: Sleepers now draw reagents from an attached chembag
del: Sleepers no longer create infinite medicines out of thin air
del: Removed the party pod. Not going to explain why other than <i>why was it in in the first place</i>
/:cl:

### CODE REVIEW

**Comment by ADepressedPyroshark** [comment]
Can you put in poisons? if no, can that be an emag feature?

**Comment by ADepressedPyroshark** [comment]
Also, @MarkSuckerberg how will this affect pod sleepers, the mining ones?

**Comment by MarkSuckerberg** [comment]
image

**Comment by MarkSuckerberg** [comment]
also the mining sleepers will have the same starting chems as the normal one, also subject to change

File: waspstation/code/modules/reagents/reagent_containers/bottle.dm [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	desc = "A small bottle. Contains kelotane, used for minor burns and skin damage."
+	list_reagents = list(/datum/reagent/medicine/kelotane = 30)
+
+/obj/item/reagent_containers/glass/bottle/antitoxin
+	name = "dylovene bottle"
+	desc = "A small bottle. Contains dylovene, used to treat minor poisoning."
+	list_reagents = list(/datum/reagent/medicine/antitoxin = 30)
+
+/obj/item/reagent_containers/glass/bottle/dexalin
+	name = "bicaridine bottle"
```
Comment (triplezeta) [inline waspstation/code/modules/reagents/reagent_containers/bottle.dm:22]: bruh

Comment (triplezeta) [inline waspstation/code/modules/reagents/reagent_containers/bottle.dm:22]: bicaridine bottle

**Review by triplezeta — CHANGES_REQUESTED** [review]

File: waspstation/code/game/machinery/Sleeper.dm [hunk 2]
```diff
@@ -13,34 +13,40 @@
...
-	var/obj/item/stock_parts/cell/cell //The sleeper's power cell.
 	var/efficiency = 1
 	var/min_health = -25
-	var/list/available_chems
+	///Possible granularity of injections
+	var/list/transfer_amounts
+	///List in which all currently dispensable reagents go
+	var/list/dispensable_reagents = list()
+
+	var/list/starting_beakers = list(
```
Comment (triplezeta) [inline waspstation/code/game/machinery/Sleeper.dm:23]: missing dexalin in your list

**Review by triplezeta — CHANGES_REQUESTED** [review]

File: waspstation/code/game/machinery/Sleeper.dm [hunk 3]
```diff
@@ -49,11 +56,11 @@
 
 	efficiency = initial(efficiency)* E
 	min_health = initial(min_health) * E
-	poweruse = max(100, 300/efficiency)
-	available_chems = list()
-	for(var/i in 1 to I)
-		available_chems |= possible_chems[i]
```
Comment (Superyodeler) [inline waspstation/code/game/machinery/Sleeper.dm:55]: Does a better manipulator still have an effect with this change?

Comment (MarkSuckerberg) [reply waspstation/code/game/machinery/Sleeper.dm:55]: Yes, it increases the amount of possible granularities for dispensing chems

Comment (Superyodeler) [reply waspstation/code/game/machinery/Sleeper.dm:55]: Matter bins affect granularity, not manipulators. It doesn't look like `I` is used for anything now that available_chems aren't a thing.

File: waspstation/code/game/machinery/Sleeper.dm [hunk 4]
```diff
@@ -105,6 +116,32 @@
 			return
 	close_machine(target)
 
+/obj/machinery/sleeper/attackby(obj/item/W, mob/living/user, params)
+	if(istype(W, /obj/item/storage/bag/chemistry))
+		. = TRUE //no afterattack
+		replace_chembag(user, W)
+	else if(chembag && istype(W, /obj/item/reagent_containers) && !(W.item_flags & ABSTRACT) && W.is_open_container())
+		forceMove(W, chembag)
```
Comment (Superyodeler) [inline waspstation/code/game/machinery/Sleeper.dm:124]: This actually puts the sleeper in the beaker, you probably should use the transferItemToLoc proc.

**Review by Superyodeler — CHANGES_REQUESTED** [review]
Found one funny bug involving a sleeper being crammed into a beaker.

**Comment by Superyodeler** [comment]
Didn't realize this had been updated, I'll test this out again.

File: waspstation/code/game/machinery/Sleeper.dm [hunk 5]
```diff
@@ -105,6 +116,32 @@
 			return
 	close_machine(target)
 
+/obj/machinery/sleeper/attackby(obj/item/W, mob/living/user, params)
+	if(istype(W, /obj/item/storage/bag/chemistry))
+		. = TRUE //no afterattack
+		replace_chembag(user, W)
+	else if(chembag && istype(W, /obj/item/reagent_containers) && !(W.item_flags & ABSTRACT) && W.is_open_container())
+		user.transferItemToLoc(W, chembag)
+		to_chat(user, "<span class='notice'>You put [W] into [src]'s [chembag].</span>")
```
Comment (Superyodeler) [inline waspstation/code/game/machinery/Sleeper.dm:125]:
```suggestion
		to_chat(user, "<span class='notice'>You put [W] into [src]'s [chembag].</span>")
		update_contents()
```

**Review by Superyodeler — CHANGES_REQUESTED** [review]
The beaker is added to the chembag correctly, but it doesn't appear on the interface until update_contents gets called.

**Review by Superyodeler — APPROVED** [review]

**Pull request merged**